lib/utils.h

```diff
@@ -21,3 +23,16 @@ inline std::string StringVectorToString(std::vector<std::string> strings) {
     result = result.substr(0, pos + 1);
   return result;
 }
+
+template <typename Func>
+std::string exceptionToString(Func func) {
```

Comment:
Style: since this is internal to the library, it should be using `snake_case`.

That said, though, I think I agree with your qualms about this util function. It might be easier to have this take the exception and format a string (i.e. basically just do what line 33 is doing here), and then the calling code would be more like 

```c++
std::string actions;

try {
  long count = this->nActions();
  for (int i = 0; i < count; ++i) {
    std::string name = this->getName(i);
    actions += " " + name;
  }
} catch (const std::exception& e)
  actions = exception_to_string(e)
}

return "Actions:" + actions;
```